docs: add more information for unittests (#716)
bareos/bareos: docs/manuals/source/DeveloperGuide/tests.rst

> The unit tests reside in ``core/src/tests`` and are compiled with the rest of Bareos if Google Test is available on your system.

Reviewer: ```suggestion
The unit tests reside in ``core/src/tests``. If Google Test is available on your system, the tests are compiled during the normal build process of Bareos.
```
